typer/core.py

```diff
@@ -419,6 +421,7 @@ def __init__(
         show_envvar: bool = False,
         # Rich settings
         rich_help_panel: Union[str, None] = None,
+        multiple_separator: Optional[str] = None,
```

Comment:
I think just `separator` would get straight to the point here, no need to preprend with "multiple" IMO

Reply:
Hmm, I agree that multiple_separator is a poor choice. It's too verbose and does not communicate it's intended usage well. How do you feel about item_separator or list_separator? I think separator may be a bit too generic. It would not be immediately clear to me that the separator keyword argument is legal only for list types.

Reply:
IMO it's not a problem if it's too generic.

Now it's only legal for list types, true, but that doesn't mean we can't give it another meaning if we find another use case for a "separator" on other types

Take for example the existing file parameters: https://github.com/tiangolo/typer/blob/3cdd682244358cf9108291ce8695a93878c46f92/typer/params.py#L54-L59

None of them have a prefix, and yet it's fine.

Same for the existing Number parameters: https://github.com/tiangolo/typer/blob/3cdd682244358cf9108291ce8695a93878c46f92/typer/params.py#L48-L51

And others like Choise or Datetime

Reply:
I see your point. I've updated the PR accordingly.